src/common/types.ts

```diff
@@ -110,12 +110,16 @@ export type RouterStoreContext = {
   location: Location;
 };
 
+type RouteComponentProps = Partial<RouteContext> & {
+  [k: string]: any;
+};
+
 export type Route = {
   path: string;
   exact?: boolean;
 
   /** The component to render on match, typed explicitly */
-  component: ComponentType<RouteContext>;
+  component: ComponentType<RouteComponentProps> | ComponentType<any>;
```

Comment:
Could this just be made into an optional generic like in the `withRouter` case? This should only need to be supported for consumers that do not use the exported `RouteComponent` component, and I would avoid the use of `any` here as a result. The default prop type just just be `RouteComponentProps` for those using the route component.